script/component-migration/index.js

```diff
@@ -60,7 +60,12 @@ function translateProps(componentString, propMap) {
  */
 function replaceTags(fileContents, newTag) {
   const unnamedClosingTags = fileContents.matchAll(
-    new RegExp(`(<${options.component}.+?(^\\s+)?\\/>;?$)`, 'gsm'),
+    new RegExp(
+      `(<${options.component}(?!.*<\\/${
+        options.component
+      }>).*?(^\\s+)?\\/>;?$)`,
+      'gsm',
+    ),
```

Comment:
Since prettier made this harder to read, the change was replacing that `.+` with the following:

```
(?!.*<\\/${options.component}>).*
```

This adds negative lookahead that will fail the match if a closing tag for the component is found.